### ISSUE TITLE
[graphiql] feat: add more root types in schema documentation

### ISSUE THREAD

**Issue by markbenliyan** [graphql/graphiql#3309]
Would be awesome to be able to add some more "parent" types to the SchemaDocumentation component, the way we have `query` `mutation` and `subscription`. 

My schema currently has a `_typemap` which gets displayed in the `All Schema Types` though the ability to categorize that would be great, like in the example below. These could map to the exisiting GraphQL types like
`GraphQLObjectType` `GraphQLEnumType``GraphQLInputObjectType` etc. I think this could be accomplished relatively easily with a prop where the User could pass in a dictionary like `{'objects': GraphQLObjectType[]}` and so on

<img width="145" alt="Screenshot 2023-07-01 at 1 14 19 AM" src="https://github.com/graphql/graphiql/assets/92900079/070d0c12-a185-4e5c-adeb-5c57373708cc">

**Comment by acao** [comment]
interesting idea! I'm going to ponder this a bit. you would expect this to be a boolean prop, yes?

**Comment by markbenliyan** [comment]
> interesting idea! I'm going to ponder this a bit. you would expect this to be a boolean prop, yes?

Yup! Or if we wanted it to be more specific maybe users can pass in the types they’d like to essentially turn into root types. 

I could look into making a PR for this personally as well. My project needs this functionality so hoping we can get it in🙏

**Comment by markbenliyan** [comment]
oh, and as a small note -- I'm imagining this would get passed in through DocExplorer so we can render the DocExplorer with this type of Schema Documentation! Clarifying in case I wasn't clear earlier

**Comment by acao** [comment]
@markbenliyan indeed! we will soon move doc explorer to it's own plugin as originally proposed in 2019, but you could start with a PR to `graphiql` and `@graphiql/react`
